### PR TITLE
cross jump

### PR DESCRIPTION
**Pull request by dannypsnl**

### CODE REVIEW

**Comment by jeapostrophe** [comment]
Neat, what exactly does it do and what remains?

**Comment by dannypsnl** [comment]
Originally, jump to the definition will jump to require xxx, now it will directly jump to cross file definition position

File: text-document.rkt [hunk 1]
```diff
@@ -257,8 +283,14 @@
      (define result
        (match decl
          [#f (json-null)]
-         [(Decl _ start end)
+         [(or (Decl #t id start end) (Decl #f id start end))
           (Location #:uri uri
+                    #:range (start/end->Range doc-text start end))]
+         [(Decl path id 0 0)
```
Comment (dannypsnl) [inline text-document.rkt:289]: we need `id` for here

File: text-document.rkt [hunk 2]
```diff
@@ -257,8 +283,14 @@
      (define result
        (match decl
          [#f (json-null)]
-         [(Decl _ start end)
+         [(or (Decl #t id start end) (Decl #f id start end))
```
Comment (dannypsnl) [inline text-document.rkt:286]: local jump will only have `#t` or `#f`

File: text-document.rkt [hunk 3]
```diff
@@ -247,6 +250,29 @@
...
+  (define ns (make-base-namespace))
+  (define-values (add-syntax done)
+    (make-traversal ns path))
+  (parameterize ([current-annotations collector]
+                 [current-namespace ns]
+                 [current-load-relative-directory src-dir])
+    (define stx (expand (with-module-reading-parameterization
+                          (λ () (read-syntax path in)))))
+    (add-syntax stx))
+  (send collector get id))
```
Comment (dannypsnl) [inline text-document.rkt:275]: get definition via id

File: doc-trace.rkt [hunk 4]
```diff
@@ -113,8 +115,9 @@
       (define new-bindings (set-add prev-bindings (cons end-left end-right)))
       (interval-map-set! sym-decls start-left start-right new-bindings)
       ;; Mapping from binding to declaration.
-      (define new-decl (Decl require-arrow? start-left start-right))
-      (interval-map-set! sym-bindings end-left end-right new-decl))
+      (unless require-arrow?
+        (define new-decl (Decl require-arrow? #f start-left start-right))
+        (interval-map-set! sym-bindings end-left end-right new-decl)))
```
Comment (dannypsnl) [inline doc-trace.rkt:120]: only record local jump since we have no path information at here

File: doc-trace.rkt [hunk 5]
```diff
@@ -99,6 +98,9 @@
                                    url-tag)
                          url))
         (interval-map-set! docs start finish (list (url->string url2) def-tag))))
+    (define/override (syncheck:add-jump-to-definition source-obj start end id filename submods)
+      (define decl (Decl filename id 0 0))
+      (interval-map-set! sym-bindings start (add1 end) decl))
```
Comment (dannypsnl) [inline doc-trace.rkt:103]: record cross-file jump here

**Comment by dannypsnl** [comment]
NOTE: https://github.com/racket/drracket/issues/498 will block this PR

**Comment by dannypsnl** [comment]
Any updates here?

**Comment by jeapostrophe** [comment]
What do you mean? Is it ready to merge?

**Comment by dannypsnl** [comment]
Yes, I think it's ready

**Pull request merged**